src/logging.cpp

```diff
@@ -249,11 +290,34 @@ namespace BCLog {
     }
 } // namespace BCLog
 
-void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& logging_function, const std::string& source_file, const int source_line)
+void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& logging_function, const std::string& source_file, const int source_line, const BCLog::LogFlags flag, const BCLog::Level level)
 {
     StdLockGuard scoped_lock(m_cs);
     std::string str_prefixed = LogEscapeMessage(str);
 
+    if ((flag != LogFlags::NONE || level != Level::None) && m_started_new_line) {
+        std::string s;
+        s += "[";
+
+        if (flag != LogFlags::NONE) {
+            auto it = LogFlagStr.find(flag);
+            if (it != LogFlagStr.end()) s += it->second;
+        }
+
+        if (flag != LogFlags::NONE && level != Level::None) {
+            // Only add separator if both flag and level are not NONE
+            s += ":";
+        }
+
+        if (level != Level::None) {
+            auto it = LevelStr.find(level);
+            if (it != LevelStr.end()) s += it->second;
```

Comment:
Wouldn't it be better to do the enum-class -> string conversion with a compile-time checked switch-case instead of an unordered map?

Reply:
Yup, you're right. Updated the PR to replace the unordered map with a helper function that uses a switch-case to return the string conversion.